packages/clay-shared/src/Portal.tsx

```diff
@@ -42,6 +42,13 @@ export const ClayPortal: React.FunctionComponent<
 				: closestParent;
 
 		if (elToMountTo && portalRef.current) {
+			if (otherProps?.className) {
+				portalRef.current.classList.add(otherProps.className);
+			}
+			if (otherProps?.id) {
+				portalRef.current.id = otherProps.id;
+			}
+
```

Comment:
Maybe we can create a function around the `document.createElement` so that we can create the element already with the attributes and we can call it directly in `useRef`. I have a feeling this would be cleaner, what do you think?

Something like that:

```ts
const createElement = (nodeName: string, attributes: Record<string, string>) => {
  const element = document.createElement(nodeName);

  Object.keys(attributes).forEach((key) => {
     const value = attributes[key];

     element.setAttribute(key, value);
  });

  return element;
}

```

Reply:
Thank @matuzalemsteles, I've done it here https://github.com/liferay/clay/pull/4067/commits/231e1640295c39b5bf5f49dd021114bad67a7b9e. Although, I was not able to type `attributes` as you mentioned above, it returned me this error: 
```
Argument of type '{ defaultChecked?: boolean | undefined; defaultValue?: string | string[] | undefined; suppressContentEditableWarning?: boolean | undefined; suppressHydrationWarning?: boolean | undefined; ... 247 more ...; onTransitionEndCapture?: TransitionEventHandler<...> | undefined; }' is not assignable to parameter of type 'Record<string, string>'.
```

And following that, I'm not sure if I can type it with something else than "any", what do you think?

Reply:
> And following that, I'm not sure if I can type it with something else than "any", what do you think?

I think that if we're using TypeScript it's to actually avoid using `any`, but that's just my opinion.

Reply:
Yeah it's always good to avoid using `any`, probably this TypeScript error is because we are doing the union for `React.HTMLAttributes<HTMLDivElement>`, we could remove this and add the types just for `id` and `className`, then it would not conflict with the `attributes` typing.

https://github.com/liferay/clay/blob/231e1640295c39b5bf5f49dd021114bad67a7b9e/packages/clay-shared/src/Portal.tsx#L16

Reply:
Hi @matuzalemsteles @julien please check commit https://github.com/liferay/clay/pull/4067/commits/4faedf64195f59851748a9d0d151157738213e32 to see if I understood you correctly. Otherwise, apologises and do not hesitate to contact me via slack. Thanks!